packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -219,6 +219,26 @@ export class ProjectLifecycleManager {
         this.ctx.emitter.toLaunchpad()
       },
       onFinalConfigLoaded: async (finalConfig: FullConfig) => {
+        if (this._currentTestingType && finalConfig.specPattern) {
+          await this.ctx.actions.project.setSpecsFoundBySpecPattern({
+            path: this.projectRoot,
+            testingType: this._currentTestingType,
+            specPattern: this.ctx.modeOptions.spec || finalConfig.specPattern,
+            excludeSpecPattern: finalConfig.excludeSpecPattern,
+            additionalIgnorePattern: finalConfig.additionalIgnorePattern,
+          })
+        }
+
+        if (this._currentTestingType === 'component') {
+          const devServerOptions = await this.ctx._apis.projectApi.getDevServer().start({ specs: this.ctx.project.specs, config: finalConfig })
```

Comment:
There is an unhandled edge-case here. If the user modifies their webpack config such that the port of the dev-server is changed, serving tests will fail. We would need to shut down and restart the server in order to fix this, but this seems like a fairly large refactor and not just ct specific.

I haven't done an audit of all config options, but there are a few options that would require a server restart such as `watchForFileChanges` or `hosts` which we currently aren't handling.

Reply:
It would be CT specific since E2E does not use the dev-server.

Also isn't the port always randomly assigned each time webpack-dev-server starts?

Reply:
So far my testing has shown that the dev-servers will default to the same port used, I only saw a port switch when explicitly changing the port in the `webpackConfig`. What I have wired up isn't fullproof, but we have [UNIFY-1696](https://cypress-io.atlassian.net/browse/UNIFY-1696) to properly handle config changes that require a server restart.